### PR TITLE
Added basic ability to fetch website for manifest

### PR DESCRIPTION
**Pull request by beaufortfrancois**
This is the beginning of more checks but for now I thought a simple fetch would be OK to iterate on it.

screenshot 2015-12-28 at 11 31 24 am

FIX https://github.com/mounirlamouri/manifest-validator/issues/4

### CODE REVIEW

**Comment by samccone** [comment]
need to take for a spin locally, however code looks LGTM

File: main.py [hunk 1]
```diff
@@ -1,14 +1,58 @@
...
+              response['websiteUrl'] = website_url
+              response['manifestUrl'] = manifest_url
+              response['content'] = manifest_result.content
+            else:
+              # TODO Check more stuff
+              response['error'] = 'Manifest %s is not HTTP 200.' % manifest_url
+          except (urlfetch.InvalidURLError, urlfetch.DownloadError) as e:
+            response['error'] = repr(e)
+      else:
+        # TODO Check more stuff
```
Comment (mounirlamouri) [inline main.py]: Check what?

File: main.py [hunk 2]
```diff
@@ -1,14 +1,58 @@
...
+          if "://" not in manifest_url:
+            manifest_url = urljoin(website_url, manifest_url)
+          try:
+            manifest_result = urlfetch.fetch(manifest_url)
+            if manifest_result.status_code == 200:
+              response['websiteUrl'] = website_url
+              response['manifestUrl'] = manifest_url
+              response['content'] = manifest_result.content
+            else:
+              # TODO Check more stuff
```
Comment (mounirlamouri) [inline main.py]: Check what?

File: main.py [hunk 3]
```diff
@@ -1,14 +1,58 @@
...
+          # Fetch web manifest.
+          manifest_url = value[0]
+          if "://" not in manifest_url:
+            manifest_url = urljoin(website_url, manifest_url)
+          try:
+            manifest_result = urlfetch.fetch(manifest_url)
+            if manifest_result.status_code == 200:
+              response['websiteUrl'] = website_url
+              response['manifestUrl'] = manifest_url
+              response['content'] = manifest_result.content
```
Comment (mounirlamouri) [inline main.py:35]: Add a TODO to check the CORS headers.

File: app.yaml [hunk 4]
```diff
@@ -3,7 +3,7 @@ api_version: 1
 threadsafe: true
 
 handlers:
-- url: /
+- url: /(fetch)?
```
Comment (mounirlamouri) [inline app.yaml]: Maybe call it "check" instead of "fetch"?

File: static/ui.js [hunk 5]
```diff
@@ -61,6 +64,30 @@ document.querySelector('input[type=file]').onchange = function() {
   };
 }
 
+document.querySelector('#check-website-url').onclick = function(e) {
+  _clearLogs();
+  _clearTips();
+
+  var websiteUrl = document.querySelector('#website-url').value;
+  var xhr = new XMLHttpRequest();
+  xhr.open('POST', '/fetch?websiteUrl=' + websiteUrl);
```
Comment (mounirlamouri) [inline static/ui.js]: Could you pass the website url in a data given that it's a POST request?

**Comment by mounirlamouri** [comment]
lgtm with the comments above. Sorry for the delay!

File: main.py (hunk 3 above)
Comment (beaufortfrancois) [inline main.py:35]: Done.

**Comment by beaufortfrancois** [comment]
I've addressed all your feedback and rebased. 
Thanks @mounirlamouri.

**Pull request merged**